skrf/time.py

```diff
@@ -287,47 +279,29 @@ def time_gate(ntwk, start: float = None, stop: float = None, center: float = Non
     window_width = abs(stop_idx - start_idx)
     window = signal.get_window(window, window_width)
 
+    nw_gated = ntwk.copy()
+
     # create the gate by padding the window with zeros
-    gate = npy.r_[npy.zeros(start_idx),
-                           window,
-                           npy.zeros(len(t) - stop_idx)]
-
-    #FFT the gate, so we have it's frequency response, aka kernel
-    kernel=fft.ifftshift(fft.fft(fft.fftshift(gate, axes=0), axis=0))
-    kernel =abs(kernel).flatten() # take mag and flatten
-    kernel=kernel/sum(kernel) # normalize kernel
-
-    out = ntwk.copy()
-
-    # conditionally delay ntwk, to center at t=0, this is
-    # equivalent to gating at center.  (this is probably very inefficient)
-    if center!=0:
-        out = out.delay(-center*1e9, 'ns',port=0,media=media)
-
-    # waste of code to handle convolve1d suck
-    re = out.s_re[:,0,0]
-    im = out.s_im[:,0,0]
-    s = convolve1d(re,kernel, mode=boundary)+\
-     1j*convolve1d(im,kernel, mode=boundary)
-    out.s[:,0,0] = s
-    # conditionally  un-delay ntwk
-    if center!=0:
-        out = out.delay(center*1e9, 'ns',port=0,media=media)
+    gate = npy.zeros_like(t)
+    gate[start_idx:stop_idx] = window
+
+    # time-domain gating
+    s_td = npy.fft.ifftshift(npy.fft.ifft(nw_gated.s[:, 0, 0]))
+    s_td_g = s_td * gate
+    nw_gated.s[:, 0, 0] = npy.fft.fft(npy.fft.fftshift(s_td_g))
```

Comment:
Suggestion: addition of `from numpy.fft import fft, ifft, fftshift, ifftshift` on top of the file to improve readability.
```Python
    s_td = ifftshift(ifft(nw_gated.s[:, 0, 0]))
    s_td_g = s_td * gate
    nw_gated.s[:, 0, 0] = fft(fftshift(s_td_g))
```

Reply:
Thanks! Done.